src/test/shell/bazel/bazel_example_test.sh

```diff
@@ -118,8 +147,6 @@ function test_native_python_with_zip() {
 
 function test_shell() {
   assert_build "//examples/shell:bin"
-  unset RUNFILES_DIR
```

Comment:
Here's a problem.

You **have to** unset these envvars to run the binary, otherwise the binary will pick up the envvars from the current environment. We don't want that.
You **mustn't** unset these envvars, otherwise you fail to run `bazel`. (`bazel` is actually a function, not a binary.)

So you can't `bazel run` a binary inside of a test, if that binary uses runfiles. Sorry.

The solution is to build the binary, then run it from `bazel-bin`, in a subshell where you unset RUNFILES_* envvars.

Reply:
Moved it into bin/bazel as discussed offline